src/main/java/hudson/plugins/dimensionsscm/DimensionsAPI.java

```diff
@@ -540,14 +556,14 @@ public boolean checkout(final long key, final String projectName, final FilePath
      * Generate changelog file.
      */
     public void createChangeSetLogs(final long key, final String projectName, final FilePath projectDir,
-            final Calendar fromDate, final Calendar toDate, final File changelogFile, final TimeZone tz,
-            final String url, final String baseline, final String requests) throws IOException {
+                                    final Calendar fromDate, final Calendar toDate, final File changelogFile, final TimeZone tz,
+                                    final String url, final String baseline, final String requests) throws IOException {
         DimensionsConnection connection = getCon(key);
         if (connection == null) {
             throw new IOException("Not connected to an SCM repository");
         }
         try {
-            List<ItemRevision> items = calcRepositoryDiffs(key, projectName, baseline, requests, projectDir, fromDate, toDate, tz);
+            List<ItemRevision> items = calcRepoDiffsWithRevisions(connection, projectName, baseline, requests, projectDir, fromDate, toDate, tz);
```

Comment:
This will get the list of "what changed" using the old method, but the "has anything changed" is done using the new method. This will lead to confusing pages in Jenkins like builds being triggered even though the "Changes" page in Jenkins says there were no changes.

Reply:
Note that if you do change the code that queries the list of changes to display, `calcRepoDiffsWithRevisions` can have non-`null` values for baseline or requests. In these cases, probably best to use the old methods for now (and only use changesets to get the list of changes to display when baselines and requests are not used).